File: ccif/src/java/other/org/himalay/ccif/nOF.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:07 EST 2016")

public  class nOF extends BinStruct implements PublicBinMsg {




    // members variables
    // featureId
    public short featureId ;
    // status
    public short status ;

    public nOF () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize featureId

        // Initialize status

    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read featureId
        {
            featureId=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read status
        {
            status=(short)(istream.readUnsignedByte());
            retVal+=1;
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;




        // write featureId
        ostream.writeByte(featureId);
        retVal +=1;
        // write status
        ostream.writeByte(status);
        retVal +=1;
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("nOF\n");
        dc.increaseIndent();
        int retVal= 0;
        // write featureId
        dc.indent();
        dc.getPs().println("featureId="+featureId+"(0x"+ Integer.toHexString(featureId) + ")" );
        // write status
        dc.indent();
        dc.getPs().println("status="+status+"(0x"+ Integer.toHexString(status) + ")" );
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for featureId
    //public short getFeatureId()
    //{
    //    return featureId ;
    //}


    // Setter for featureId
    //public void setFeatureId(short val)
    //{
    //    this.featureId= val;
    //}
    // Getter for status
    //public short getStatus()
    //{
    //    return status ;
    //}


    // Setter for status
    //public void setStatus(short val)
    //{
    //    this.status= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/nOF.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "nOF.h"

#ifdef org_himalay_ccif_nOF__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
nOF::nOF() {
    references= 0;


    // featureId
    featureId= (ui8)0;
    // status
    status= (ui8)0;
}
// Destructor
nOF::~nOF() {
    // featureId

    // status

}

int nOF::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // featureId
    {
        featureId=(istream.readUI8());
        retVal+=1;
    }
    // status
    {
        status=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int nOF::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read featureId
    {
        featureId=(istream.readUI8());
        retVal+=1;
    }
    // read status
    {
        status=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int nOF::write(DataOutputStream& ostream) { // throws IOException



    int retVal= 0;
    // write featureId
    ostream.writeUI8(featureId);
    retVal +=1;
    // write status
    ostream.writeUI8(status);
    retVal +=1;

    return retVal;
}

int nOF::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("nOF\n");
    dc.increaseIndent();
    int retVal= 0;
    // write featureId
    dc.indent();
    dc.getPs().print("featureId=");
    dc.getPs().println((long)featureId);
    // write status
    dc.indent();
    dc.getPs().print("status=");
    dc.getPs().println((long)status);
    dc.decreaseIndent();
    return retVal;
}


// Getter for featureId
//ui8 nOF::getFeatureId()
//{
//return this->featureId;
//}

// Setter for featureId
//void nOF::setFeatureId(ui8 val)
//{
//this->featureId= val;
//}

// Getter for status
//ui8 nOF::getStatus()
//{
//return this->status;
//}

// Setter for status
//void nOF::setStatus(ui8 val)
//{
//this->status= val;
//}

int nOF::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_nOF__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::nOF* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::nOF* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code